src/utils/storage.ts

```diff
@@ -0,0 +1,54 @@
+interface Store {
+  beeApiUrl: string
+}
+
+type StoreKey = keyof Store
+
+export function getItem<T extends StoreKey>(key: T): Promise<Store[T]> {
+  return new Promise(resolve => {
+    chrome.storage.local.get(key, items => {
+      console.log('items', items)
+      resolve(items[key])
+    })
+  })
+}
+
+export async function setItem<T extends StoreKey>(key: T, value: Store[T]): Promise<void> {
+  await new Promise(resolve => {
+    chrome.storage.local.set({ [key]: value }, () => resolve)
+  })
+}
+
+type StoreChangeCallback<T> = (newValue: T, oldValue: T, namespace: 'sync' | 'local' | 'managed', key: string) => void
+
+export class StoreObserver {
```

Comment:
This to me looks like an event emitter and potentially singleton. I'd suggest to either use default `EventEmitter` or even [`emittery`](https://github.com/sindresorhus/emittery) which can handle async events in both async and blocking mode.